manifests/server.pp

```diff
@@ -13,7 +13,7 @@
   String $package_name                                                          = $prometheus::package_name,
   String $package_ensure                                                        = $prometheus::package_ensure,
   String $config_dir                                                            = $prometheus::config_dir,
-  Stdlib::Absolutepath $localstorage                                            = $prometheus::localstorage,
+  Variant[Stdlib::Absolutepath, Boolean[false]] $localstorage                   = $prometheus::localstorage,
```

Comment:
The Boolean data type accepts no parameters. https://puppet.com/docs/puppet/7/lang_data_boolean.html

Reply:
Actually, this works as intended:

```bash
$ mkdir -p modules/foo/manifests/
$ cat <<EOF > modules/foo/manifests/init.pp
class foo (
  Variant[Stdlib::Absolutepath, Boolean[false]] \$localstorage,
) { }
EOF
$ git clone https://github.com/puppetlabs/puppetlabs-stdlib modules/stdlib
```

```bash
$ puppet apply --modulepath modules/ -e 'include foo'
Error: Evaluation Error: Error while evaluating a Function Call, Class[Foo]: expects a value for parameter 'localstorage' (line: 1, column: 1) on node localhost
$ puppet apply --modulepath modules/ -e "include class {"foo": localstorage => '/var/lib/prom' }"
Notice: Compiled catalog for localhost in environment production in 0.03 seconds
Notice: Applied catalog in 0.01 seconds
$ puppet apply --modulepath modules/ -e "include class {"foo": localstorage => false }"
Notice: Compiled catalog for localhost in environment production in 0.03 seconds
Notice: Applied catalog in 0.01 seconds
$ puppet apply --modulepath modules/ -e "include class {"foo": localstorage => true }"
Error: Evaluation Error: Error while evaluating a Resource Statement, Class[Foo]: parameter 'localstorage' expects a value of type Stdlib::Absolutepath = Variant[Stdlib::Windowspath = Pattern[/\A(([a-zA-Z]:[\\\/])|([\\\/][\\\/][^\\\/]+[\\\/][^\\\/]+)|([\\\/][\\\/]\?[\\\/][^\\\/]+)).*\z/], Stdlib::Unixpath = Pattern[/\A\/([^\n\/\0]+\/*)*\z/]] or Boolean[false], got Boolean[true] (line: 1, column: 9) on node localhost
```

Reply:
Oh, well, that's weirdly undocumented.